sphinx/ext/mathbase.py

```diff
@@ -34,16 +34,17 @@ def wrap_displaymath(math, label, numbering):
     for i, part in enumerate(parts):
         if not part.strip():
             continue
-        if label is not None and i == 0:
-            ret.append('\\begin{split}%s\\end{split}' % part +
-                       (label and '\\label{'+label+'}' or ''))
-        else:
-            ret.append(r'\begin{split}%s\end{split}' % part)
-            if not numbering:
-                ret.append(r'\notag')
+        ret.append(r'\begin{split}%s\end{split}' % part)
     if not ret:
         return ''
-    return '\\begin{gather}\n' + '\\\\'.join(ret) + '\n\\end{gather}'
+    if label is not None or numbering:
+        env_begin = r'\begin{align}\label{%s}' % label
```

Comment:
`\label{None}` will be generated here if label is None and numbering(`math_number_all`) is True.